test/edu-benefits/utils/veteran.unit.spec.js

```diff
@@ -41,6 +41,49 @@ describe('veteranToApplication', () => {
 
     expect(applicationData.toursOfDuty[0].fromDate).to.equal('1996-09-06');
   });
+  it('converts date ranges to iso', () => {
+    const formData = createVeteran();
+    formData.toursOfDuty.push({
+      serviceBranch: makeField('army'),
+      dateRange: {
+        from: {
+          month: makeField('9'),
+          day: makeField('6'),
+          year: makeField('1996')
+        },
+        to: {
+          month: makeField('9'),
+          day: makeField('6'),
+          year: makeField('1997')
+        }
+      }
+    });
+    const applicationData = JSON.parse(veteranToApplication(formData));
+
+    expect(applicationData.toursOfDuty[0].dateRange.from).to.equal('1996-09-06');
+    expect(applicationData.toursOfDuty[0].dateRange.to).to.equal('1997-09-06');
+  });
+  it('removes partial date ranges', () => {
```

Comment:
The only thing I'm a little worried about is removing partial dates just for the sake of passing our schema validations. From an education services perspective, partial date information is still more helpful than no date information, right? Is there a reason why we can't submit partial date information?